lib/utils.pm

```diff
@@ -1022,4 +1023,27 @@ sub svirt_host_basedir {
     return get_var('VIRSH_OPENQA_BASEDIR', '/var/lib');
 }
 
+sub prepare_ssh_localhost_key_login {
+    my ($source_user) = @_;
+    # in case localhost is already inside known_hosts
+    if (script_run('test -e ~/.ssh/known_hosts') == 0) {
+        assert_script_run('ssh-keygen -R localhost');
+    }
+
+    # generate ssh key
+    if (script_run('! test -e ~/.ssh/id_rsa') == 0) {
+        assert_script_run('ssh-keygen -t rsa -N "" -f ~/.ssh/id_rsa');
+    }
+
+    # add key to authorized_keys of root
+    if ($source_user eq 'root') {
+        assert_script_run('mkdir -p ~/.ssh');
```

Comment:
why do you check for creating the dir at this point? The two previous operation would fail if it didn't exist before?

Reply:
That's right - I removed that line.